dev_guide/application_lifecycle/promoting_applications.adoc

```diff
@@ -0,0 +1,781 @@
+** are injected into containers for application use.
+** dictate how {product-title} manages containers and pods.
+
+In examining how to promote applications in {product-title}, this topic will:
+
+* Elaborate on these new artifacts introduced to the application definition.
+* Describe how you can demarcate the different environments for your application promotion pipeline.
+* Discuss methodologies and tools for managing these new artifacts.
+* Provide examples that apply the various concepts, constructs, methodologies, and tools to application promotion.
+
+[[dev-guide-promoting-applications-components]]
+== Application Components
+
+[[dev-guide-promoting-applications-api-objects]]
+=== API Objects
+
+With regard to {product-title} and Kubernetes resource definitions (the items
+newly introduced to the application inventory), there are a couple of key design
+points for these API objects that are relevant to revisit when considering the
+topic of application promotion.
+
+First, as highlighted throughout {product-title} documentation, every API object
+can be expressed via either JSON or YAML, making it easy to manage these
+resource definitions via traditional source control and scripting.
+
+Also, the API objects are designed such that there are portions of the object
+which specify the desired state of the system, and other portions which reflect
+the status or current state of the system. This can be thought of as inputs and
+outputs. The input portions, when expressed in JSON or YAML, in particular are
+items that fit naturally as source control managed (SCM) artifacts.
+
+[NOTE]
+====
+Remember, the input or specification portions of the API objects can be totally
+static or dynamic in the sense that
+xref:../templates.adoc#writing-parameters[variable substitution via template
+processing] is possible on instantiation.
+====
+
+The result of these points with respect to API objects is that with their
+expression as JSON or YAML files, you can treat the configuration of the
+application  as code.
+
+Conceivably, almost any of the API objects may be considered an application
+artifact by your organization. Listed below are the objects most commonly
+associated with deploying and managing an application:
+
+BuildConfigs:: This is a special case resource in the context of application
+promotion. While a `BuildConfig` is certainly a part of the application,
+especially from a developer's perspective, typically the `BuildConfig` is not
+promoted through the pipeline. It produces the `Image` which will get promoted
+(along with other items) through the pipeline.
+Templates:: In terms of application promotion, `Templates` can serve as the
+starting point for setting up resources in a given staging environment,
+especially with the parameterization capabilities. Additional post-instantiation
+modifications are very conceivable though when applications move through a
+promotion pipeline. See the examples section below for more on this.
```

Comment:
See the xref:dev-guide-promoting-applications-scenarios[examples] section for more on this.

(Remove the word "below", but include the direct link.)